covid19-dashboard_NEW/main.py

```diff
@@ -0,0 +1,360 @@
+                # It is the population.
+                if 'measuredValue' not in observation:
+                    continue
+                # Store the population.
+                output[geo_id] = observation['measuredValue']
+
+    # NYT exceptions.
+    # New York City.
+    output['geoId/3651000'] = 8_399_000
+    # Kansas City.
+    output['geoId/2938000'] = 491_918
+
+    return output
+
+
+def get_places(return_counties=False):
+    """
+    Returns a list of places in the United States.
+    :param return_counties: whether to return information
+    for counties+states or just states.
+    :return: geoId -> (name, containedIn, typeOfRegion)
+    """
+
+    # Get US State data.
+    response: dict = send_request(DC_SERVER + "node/places-in", {
+        "dcids": ["country/USA"],
+        "placeType": "State"
+    },
+                                  api_key=API_KEY)
+
+    # Get the names for the US State data.
+    state_dcids: list = [x['place'] for x in response]
+    response: dict = send_request(DC_SERVER + "node/property-values", {
+        "dcids": state_dcids,
+        "property": "name"
+    },
+                                  api_key=API_KEY)
+    states: dict = {}
+
+    # For every geoId, store (name, containedIn, typeOfRegion).
+    for geo_id in response:
+        if 'out' in response[geo_id] and response[geo_id]['out']:
+            if 'value' in response[geo_id]['out'][0]:
+                # Store the data for the geoId.
+                name = response[geo_id]['out'][0]['value']
+                states[geo_id] = (name, 'country/USA', "State")
+
+    # Get US County data.
+    response: dict = send_request(DC_SERVER + "node/places-in", {
+        "dcids": state_dcids,
+        "placeType": "County"
+    },
+                                  api_key=API_KEY)
+
+    # Keep track of what county each state belongs to.
+    county_to_state: dict = {}
+    for value in response:
+        if 'place' in value and 'dcid' in value:
+            county_geo_id: str = value['place']
+            belongs_to_geo_id: str = value['dcid']
+            county_to_state[county_geo_id]: str = belongs_to_geo_id
+
+    # Get the name of all counties
+    response: dict = send_request(DC_SERVER + "node/property-values", {
+        "dcids": list(county_to_state.keys()),
+        "property": "name"
+    },
+                                  api_key=API_KEY)
+
+    # Store all the counties.
+    counties: dict = {}
+
+    # If param return_counties == True, get county data.
+    if return_counties:
+        for geo_id in response:
+            if 'out' in response[geo_id] and response[geo_id]['out']:
+                if 'value' in response[geo_id]['out'][0]:
+                    name: str = response[geo_id]['out'][0]['value']
+                    # Get rid of trailing strings in names such as County.
+                    for region_type in REGION_TYPES:
+                        name: str = name.replace(" " + region_type, "")
```

Comment:
you can just use rstrip() or strip() here. leaving the whitespace shouldn't effect display though.

btw, do we really include trailing spaces in our county names? that's in the dc graph now? please file a bug in the data repo so we fix it in the graph for everyone.

Reply:
The KG has several alternative names including different languages, so it does contain "Miami-Dade" alone, but I need to figure out how to request it using APIs. Will look into it.